Update the GOES 16 and 17 readers

bbakernoaa opened the pull request:
fix: This adds a proper reader for the NOAA GOES 16 and GOES 17 L1 and L2 products.  It also allows for reading of the GOES data both locally or from the amazon S3 server.

bbakernoaa committed:
Update the GOES 16 and 17 readers

fix: This adds a proper reader for the NOAA GOES 16 and GOES 17 L1 and L2 products.  It also allows for reading of the GOES data both locally or from the amazon S3 server.

pep8speaks commented:
Hello @bbakernoaa! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`monetio/sat/__init__.py`](https://github.com/noaa-oar-arl/monetio/blob/e06b352df0542f5deba8a182626e96bcdfa0e49e/monetio/sat/__init__.py):

> [Line 3:80](https://github.com/noaa-oar-arl/monetio/blob/e06b352df0542f5deba8a182626e96bcdfa0e49e/monetio/sat/__init__.py#L3): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (86 > 79 characters)

* In the file [`monetio/sat/goes.py`](https://github.com/noaa-oar-arl/monetio/blob/e06b352df0542f5deba8a182626e96bcdfa0e49e/monetio/sat/goes.py):

> [Line 10:80](https://github.com/noaa-oar-arl/monetio/blob/e06b352df0542f5deba8a182626e96bcdfa0e49e/monetio/sat/goes.py#L10): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (110 > 79 characters)
> [Line 78:80](https://github.com/noaa-oar-arl/monetio/blob/e06b352df0542f5deba8a182626e96bcdfa0e49e/monetio/sat/goes.py#L78): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)
> [Line 79:80](https://github.com/noaa-oar-arl/monetio/blob/e06b352df0542f5deba8a182626e96bcdfa0e49e/monetio/sat/goes.py#L79): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 108:80](https://github.com/noaa-oar-arl/monetio/blob/e06b352df0542f5deba8a182626e96bcdfa0e49e/monetio/sat/goes.py#L108): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 127:80](https://github.com/noaa-oar-arl/monetio/blob/e06b352df0542f5deba8a182626e96bcdfa0e49e/monetio/sat/goes.py#L127): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (97 > 79 characters)
> [Line 151:80](https://github.com/noaa-oar-arl/monetio/blob/e06b352df0542f5deba8a182626e96bcdfa0e49e/monetio/sat/goes.py#L151): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (87 > 79 characters)
> [Line 164:80](https://github.com/noaa-oar-arl/monetio/blob/e06b352df0542f5deba8a182626e96bcdfa0e49e/monetio/sat/goes.py#L164): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (83 > 79 characters)
> [Line 189:80](https://github.com/noaa-oar-arl/monetio/blob/e06b352df0542f5deba8a182626e96bcdfa0e49e/monetio/sat/goes.py#L189): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (89 > 79 characters)



##### Comment last updated at 2020-06-18 14:03:20 UTC

bbakernoaa commented:
The datasets can be easily be used like so

```python
import monetio as mio
mio.goes.open_dataset(date='2020-05-10 22:00', satellite='16', product='ABI-L2-AODF')
```

bbakernoaa committed:
fix: fixed infinite latitude and longitude

There was an issue that the files had in that there are some points in the files that are not actually on the earth due to converage and array sizes.  This can cause problems when plotting as pcolormesh doesn't accept masked or non-finite valued arrays.  I simply filled the inf values with 1e30 so that they plot.

bbakernoaa commented:
@amcz @aring1988 Have either of you had time to look at this?
